description: 'ptrace: Fix ->ptracer_cred handling for PTRACE_TRACEME'
references:
- https://bugzilla.suse.com/show_bug.cgi?id=1140671
- https://bugs.chromium.org/p/project-zero/issues/detail?id=1903
- https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-13272
- http://packetstormsecurity.com/files/153663/Linux-PTRACE_TRACEME-Broken-Permission-Object-Lifetime-Handling.html
- https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=6994eefb0053799d2e07cd140df6c2ea106c41ee
- https://bugzilla.redhat.com/show_bug.cgi?id=1730895
- https://usn.ubuntu.com/usn/usn-4093-1
- https://usn.ubuntu.com/usn/usn-4094-1
- https://usn.ubuntu.com/usn/usn-4095-1
- https://usn.ubuntu.com/usn/usn-4117-1
- https://usn.ubuntu.com/usn/usn-4118-1
comments:
  debian/carnil: |-
    64b875f7ac8a ("ptrace: Capture the ptracer's creds not
    PT_PTRACE_CAP") which is in 4.10 was backported to older stable
    series as well, got included in v4.9.32
reporters:
- Jann Horn
introduced-by:
  cip/4.4: [1c1f15f8ebfbd5042883a1c9ae4b18a6299c9c5f]
  cip/4.4-rt: [1c1f15f8ebfbd5042883a1c9ae4b18a6299c9c5f]
  mainline: [64b875f7ac8a5d60a4e191479299e931ee949b67]
  stable/3.16: [d55a94ed03a24794d47f80d5300825f6c095a0a7]
  stable/4.4: [1c1f15f8ebfbd5042883a1c9ae4b18a6299c9c5f]
  stable/4.9: [e747b4ae3b6bca205d82e86366e140cdcbfb7731]
fixed-by:
  cip/4.19: [54435b7fff7bfb9515cc457b71c3734c1c3fff76]
  cip/4.19-rt: [54435b7fff7bfb9515cc457b71c3734c1c3fff76]
  cip/4.4: [54562d2b0be5c120b01c8bb94baef1aca37fd329]
  cip/4.4-rt: [54562d2b0be5c120b01c8bb94baef1aca37fd329]
  mainline: [6994eefb0053799d2e07cd140df6c2ea106c41ee]
  stable/3.16: [d5d5bd909a4f03f132ee3fd3f6f0568c8344eee5]
  stable/4.14: [bf71ef9655d25e8b275ec6ed649b6bd719231ddc]
  stable/4.19: [54435b7fff7bfb9515cc457b71c3734c1c3fff76]
  stable/4.4: [54562d2b0be5c120b01c8bb94baef1aca37fd329]
  stable/4.9: [d8b99303da935228a33fa7656a964adfb33e271b]
  stable/5.1: [49887fc3f2a7e6185777af3a9c96095aeb2bce1b]
